relentless/optimize/method.py

```diff
@@ -121,11 +131,144 @@ def abs_tol(self, value):
         else:
             self._abs_tol = abs_tol
 
+class LineSearch:
+    r"""Line search algorithm.
+
+    For an :class:`~relentless.optimize.ObjectiveFunction` :math:`f\left(\mathbf{x}\right)`,
+    the line search algorithm seeks to take a single, optimally-sized step towards
+    the minimum of the function. (Does not necessarily converge to minimum.)
+
+    A search interval :math:`\mathbf{d}` is specified, and normalized to :math:`\hat{\mathbf{d}}`:
+
+    .. math::
+
+        \mathbf{d} = \mathbf{x}_{end} - \mathbf{x}_{start}
+
+        \hat{\mathbf{d}} = \frac{\mathbf{d}}{\lVert\mathbf{d}\rVert}
+
+    Then, a 'target' value :math:`t` is defined as:
+
+    .. math::
+
+        t = -\hat{\mathbf{d}} \cdot \nabla{f\left(\mathbf{x}\right)}
+
+    The target at the start of the search interval is always negative. If the
+    target is negative (or within the tolerance) at the end of the search
+    interval, then the maximum step size is acceptable and the algorithm steps
+    to the end of the search interval. If the target is positive (outside of the
+    tolerance) at the end of the search interval, then the algorithm iteratively
+    computes a new step size by linear interpolation within the search interval
+    until the target at the new location is minimized to within the tolerance.
+
```

Comment:
I think we might also want to add a short note saying that this is applying the strong Wolfe criterion and link the wiki page, like we discussed on slack. We might also mention that we are not doing anything with criteria one.